tools/correlation_analysis/abims_correlation_analysis.xml

```diff
@@ -0,0 +1,322 @@
+                            <option value="pearson" selected="true">pearson</option>
+                            <option value="kendall">kendall</option>
+                            <option value="spearman">spearman</option>
+                         </param>
+                         <param name="param_cytoscape" type="float" label="Cytoscape correlation threshold" value="0.75" help="Choose a threshold value for selecting metabolites that will be exported to a cytoscape sif format" />
+                    </when>
+                </conditional>
+            </when>
+
+            <when value="select_input_other">
+                <param name="user_matrix_corr" type="data" label="Your table file (tabular format)" format="tabular" help="Your metabolites (variables) intensity table file (tabular format)" />
+                <param name="corr_method" type="select" label="Choice of the correlation method" help="" >
+                    <option value="pearson" selected="true">pearson</option>
+                    <option value="kendall">kendall</option>
+                    <option value="spearman">spearman</option>
+                </param>
+                <param name="param_cytoscape" type="float" label="Cytoscape correlation threshold" value="0.75" help="Threshold value for selecting edges (i.e. correlations) that will be exported to the Cytoscape sif format file" />
+            </when>
+        </conditional>
+    </inputs>
+
+    <outputs>
+        <data name="sorted_table" format="tabular" from_work_dir="sorted_table.tsv" label="sorted_variableMetadata.tsv">
+             <filter>(cond_input_type['select_input_type'] == 'select_input_from_w4m')</filter>
+        </data>
+        <data name="correlation_matrix_selected" format="tabular" from_work_dir="correlation_matrix_selected.tsv" label="correlation_matrix_selected.tsv">
+            <filter>(cond_input_type['select_input_type'] == 'select_input_from_w4m' and cond_input_type['cond_function']['select_funtion']== 'sort_and_corr' and cond_input_type['cond_function']['corrdel']== '1' )</filter>
+        </data>
+        <data name="siff_table" format="tabular" from_work_dir="siff_table.tsv" label="sif_table.tsv">
+            <filter>(cond_input_type['select_input_type'] == 'select_input_from_w4m' and cond_input_type['cond_function']['select_funtion']== 'sort_and_corr' and cond_input_type['cond_function']['corrdel']== '1' )</filter>
+        </data>
+        <data name="correlation_matrix_user" format="tabular" from_work_dir="correlation_matrix.tsv" label="correlation_matrix.tsv">
+            <filter>(cond_input_type['select_input_type'] == 'select_input_other')</filter>
+        </data>
+        <data name="siff_table_user_user" format="tabular" from_work_dir="siff_table.tsv" label="sif_table.tsv">
+            <filter>(cond_input_type['select_input_type'] == 'select_input_other')</filter>
+        </data>
+    </outputs>
+
+    <tests>
+        <test>
```

Comment:
you can add here a `expect_num_outputs` https://docs.galaxyproject.org/en/latest/dev/schema.html#id78

With this, you can test your filters on outputs.

Reply:
Done